# Assignment 3: Finish z-test lab and turn it in as a notebook

1. Formulate the hypothesis (Null and Alternative) to be tested in words and formula
2. Download of the data (which is in [https://github.com/fedhere/PUI2017_fb55/blob/master/Lab3_fb55/times.txt] - get raw data)
3. Calculate the z statistics (with the given formula and data processed from the data file)
4. Compare the statistics with the significance threshold and the conclusions about the Null Hypothesis

A new bus route for line X8 is implemented. MTA wants to know if it improves commute time (travel time at peak hours). They know what the mean travel time used to be, and measure the new travel time 100 times. 

Told ~ $N(\mu = 36, \sigma = 6)$

### Question
Is the new bus route improving the commute?

### Null Hypothesis
The commute time is on average the same or longer with the new bus route as it was before.

$T_n >= T_o$

### Alternative Hypothesis
The commute time is shorter with the new bus route as it was before.

$T_n < T_o$

In [1]:
from __future__ import print_function
import pandas as pd
import matplotlib.pyplot as pl
import numpy as np

%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
mean_old = 36.
std_old = 6.

In [3]:
times_new = pd.read_csv('https://raw.githubusercontent.com/fedhere/PUI2017_fb55/'
                 + 'master/Lab3_fb55/times.txt', header=None).values.squeeze()

In [4]:
mean_new = np.mean(times_new)
std_new = np.std(times_new)
N = len(times_new)

print('The mean of the times of the new bus route is {:.4f} and the standard deviation is {:.4f}'\
     .format(mean_new, std_new))

The mean of the times of the new bus route is 34.4662 and the standard deviation is 7.1015


# $Z = \frac{\mu_{pop} - \mu_{sample}}{\sigma / \sqrt{N}}$

In [5]:
z = (mean_old-mean_new) / (std_old/math.sqrt(N))

print('The z-test is {:.4f}'.format(z))

The z-test is 2.5564


### Since the z-test value 2.5564 is greater than 2, it means that we are farther away from $2\sigma$ from the old mean. Therefore, we have rejected our NULL HYPOTHESIS.

### The new bus route does improve the commute.